### PR TITLE
refactor(vault): simplify code and fix tests and utils by auth

### PR DESCRIPTION
**Pull request by guimroque**

### CODE REVIEW

File: packages/sdk/src/vault/Vault.ts [hunk 1]
```diff
@@ -116,89 +97,21 @@ export class Vault extends Predicate<[]> implements IVault {
...
+    switch (
+      _params.type // todo: formatt to include api and auth
+    ) {
+      case ECreationtype.IS_OLD:
+        return new Vault(_params.payload);
+      case ECreationtype.IS_NEW:
+        const vault = new Vault(_params.payload);
+        !!vault.api && (await vault.createOnService());
+        return vault;
+      case ECreationtype.ERROR:
```
Comment (luisburigo) [inline packages/sdk/src/vault/Vault.ts:111]: Para que serve o `ECreationtype.ERROR`?

File: packages/sdk/src/vault/Vault.ts [hunk 2]
```diff
@@ -116,89 +97,21 @@ export class Vault extends Predicate<[]> implements IVault {
...
+        throw new Error(_params.payload);
+      default:
+        throw new Error('Tipo não reconhecido');
```
Comment (luisburigo) [inline packages/sdk/src/vault/Vault.ts:114]: O erro deveria ser em ingles?

File: packages/sdk/src/vault/Vault.ts (hunk 1 above)
Comment (guimroque) [reply packages/sdk/src/vault/Vault.ts]: Redundant, removed ✅

File: packages/sdk/src/vault/Vault.ts (hunk 2 above)
Comment (guimroque) [reply packages/sdk/src/vault/Vault.ts]: Changed

**Review by luisburigo — APPROVED** [review]

**Pull request merged**